docs/run_test/troubleshoot_failures.rst

```diff
@@ -0,0 +1,184 @@
+Troubleshoot Test Failures
+=======================
+
+-  `Overview <#overview>`__
+-  `Test results <#test-results>`__
+-  `Console output <#console-output>`__
+-  `Log Folder Structure <#log-folder-structure>`__
+
+Overview
+--------
+
+To understand a test failure, the recommended troubleshooting path is: 
```

Comment:
Please explain more on why we recommend this. Because the test results should be able to troubleshoot most issues, and it's simple to read and get.

Reply:
I added why the user should focus on test results directly in the overview

Reply:
I added more detail for user on recommended troubleshoot order in terms of ease, time and cost